src/Utilities/CleverDicts.jl

```diff
@@ -0,0 +1,231 @@
+module CleverDicts
+
+# The following two functions are overloaded for `MOI.VariableIndex` here
+# it is the original use-case for `CleverDict`, and it would be type-piracy for
+# solvers using `CleverDicts` to implement it themselves.
+
+import MathOptInterface
+
+function index_to_key(::Type{MathOptInterface.VariableIndex}, index::Int)
+    return MathOptInterface.VariableIndex(index)
+end
+
+key_to_index(key::MathOptInterface.VariableIndex) = key.value
```

Comment:
@blegat: I've implemented the methods to use `CleverDict` with `MOI.VariableIndex` since it would be type-piracy for solvers to implement the methods. Any ideas for better places to put them? Or is here okay?

Reply:
No this seams appropriate. We it would make even more sense if we add `const CleverVariableDict{V} = CleverDict{MOI.VariableIndex, V}`